### ISSUE TITLE
VPC is created multiple-time

### ISSUE THREAD

**Issue by johanneswuerbach** [kubernetes-sigs/cluster-api-provider-aws#103]
Not sure whether this is due to https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/98, but the error in https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/102 caused the aws-cluster-controller to create VPC until the max VPC limit had been reached.

Assumably the VPC Id should be stored into the cluster spec once successfully created to prevent recreation.

**Comment by vincepri** [comment]
The describe method uses the VPC id that in from the Network configuration, if this is empty, the VPC is created. What's probably happening is that the id is never stored and it goes into a loop trying to recreate the VPC multiple times. 

We need to find a way to persist important information in the Cluster actuator and use tags to discover pre-created resources based on a cluster identifier.

**Comment by chuckha** [comment]
fixed by #106

**Comment by johanneswuerbach** [comment]
I don't think so as #106 won't be able to resume a errored out network reconciliation and probably recreate the VPC.

Instead I think tagging and discovery based on that tags might be a better safe-guard to also handle NPE and other failures between creating resources and persisting the state.

I've implemented something basic in https://github.com/virtual-kubelet/virtual-kubelet/blob/master/providers/aws/helpers_test.go and would be happy to take a try at that.

**Comment by johanneswuerbach** [comment]
Yes, I also meant both should be done. Having the ids of the created resources inside the spec is also really helpful for debugging and not just correctness.

**Comment by vincepri** [comment]
I'd rephrase the issue as a future improvement, the immediate bug (loop) was fixed in #106. I agree though that we should do both longer term, given that we can gather a cluster id somewhere. I think we also haven't standardized on tagging / identifiers yet.

**Comment by johanneswuerbach** [comment]
I created an example implementation of tagging the VPC and trying to discover it based on said tag, comments more then welcome https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/112
